selfies/encoder.py

```diff
@@ -184,29 +184,37 @@ def _fragment_to_selfies(mol, bond_into_root, root,
                 bond_into_curr, curr = bond, bond.dst
 
             else:
+                # start, end are so we can go back and
+                # correct offset from branch symbol in
+                # branch tokens
+                start = len(attribution_maps)
                 branch = _fragment_to_selfies(
                     mol, bond, bond.dst, attribution_maps, len(derived))
                 Q_as_symbols = get_selfies_from_index(len(branch) - 1)
                 branch_symbol = "[{}Branch{}]".format(
                     _bond_to_selfies(bond, show_stereo=False),
                     len(Q_as_symbols)
                 )
-
-                derived.append(branch_symbol)
-                attribution_maps.append(AttributionMap(
-                    len(derived) - 1 + attribution_index,
-                    branch_symbol, mol.get_attribution(bond)))
+                end = len(attribution_maps)
                 for symbol in Q_as_symbols:
                     derived.append(symbol)
                     attribution_maps.append(AttributionMap(
                         len(derived) - 1 + attribution_index,
                         symbol, mol.get_attribution(bond)))
+
+                # account for branch symbol because it is inserted after
+                for i in range(start, end):
+                    attribution_maps[i].index += len(Q_as_symbols) + 1
+                derived.append(branch_symbol)
```

Comment:
Sorry for the delay! I believe the reason the tests are failing might be the moving of this line, since it swaps the order of SELFIES index and branch symbols.